stock_picking_mass_assign/wizard/check_assign_all.py

```diff
@@ -27,22 +27,99 @@ class check_assign_all(orm.TransientModel):
+        wizard = self.browse(cr, uid, wizard_id, context=context)
+
+        # Get confirmed pickings
         domain = [('type', '=', 'out'),
                   ('state', '=', 'confirmed'),
                   ('id', 'in', picking_ids)]
-        picking_ids = picking_obj.search(cr, uid, domain,
-                                         order='min_date',
-                                         context=context)
-        picking_obj.check_assign_all(cr, uid, picking_ids, context=context)
-        return {'type': 'ir.actions.act_window_close'}
+        confirmed_picking_ids = picking_obj.search(
+            cr, uid, domain, order='min_date', context=context)
+
+        # Assign all picking if asked
+        if wizard.check_availability and confirmed_picking_ids:
+            picking_obj.check_assign_all(
+                cr, uid, confirmed_picking_ids, context=context)
+
+        # Force availability if asked
+        if wizard.force_availability and confirmed_picking_ids:
+            picking_obj.force_assign(cr, uid, confirmed_picking_ids)
+
+        # Get all pickings ready to deliver
+        domain = [('type', '=', 'out'),
+                  ('state', '=', 'assigned'),
+                  ('id', 'in', picking_ids)]
+        assigned_picking_ids = picking_obj.search(
+            cr, uid, domain, order='min_date', context=context)
+
+        # Process all pickings if asked
+        if wizard.process_picking and assigned_picking_ids:
+            for picking in picking_obj.browse(
+                    cr, uid, assigned_picking_ids, context=context):
+                ctx = context.copy()
+                ctx['active_ids'] = [picking.id]
+                process_wizard_id = process_wizard_obj.create(
+                    cr, uid, {}, context=ctx)
+                process_wizard_obj.do_partial(
+                    cr, uid, [process_wizard_id], context=context)
+
+        # Get all pickings ready to invoice
+        domain = [('type', '=', 'out'),
+                  ('invoice_state', '=', '2binvoiced'),
+                  ('id', 'in', picking_ids)]
+        to_invoice_picking_ids = picking_obj.search(
+            cr, uid, domain, order='min_date', context=context)
+
+        # Invoice all pickings if asked
+        if wizard.create_invoice and to_invoice_picking_ids:
+            # return the regular wizard to invoice many pickings
+            ctx = context.copy()
+            ctx['active_ids'] = to_invoice_picking_ids
+            ctx['active_model'] = 'stock.picking.out'
+            return {
+                'name': _('Stock Invoice Onshipping'),
+                'view_type': 'form',
+                'view_mode': 'form',
+                'res_model': 'stock.invoice.onshipping',
+                'type': 'ir.actions.act_window',
+                'target': 'new',
+                'context': ctx,
+            }
+        else:
+            # close the wizard
+            return {'type': 'ir.actions.act_window_close'}
```

Comment:
In fact, I wrote a "return True". Otherwise, I have a warning in a log.
openerp.osv.osv: The method check of the object stock.picking.check.assign.all can not return `None` !